File: libs/common/src/typegraph/runtimes/temporal.rs
// Copyright Metatype OÜ, licensed under the Elastic License 2.0.
// SPDX-License-Identifier: Elastic-2.0

use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Clone, Debug)]
pub struct TemporalRuntimeData {
    pub name: String,
    pub host_secret: String,
    pub namespace_secret: Option<String>,
    // pub task_queue_secret: Option<String>,
}

File: src/typegate/tests/runtimes/temporal/temporal.ts
// Copyright Metatype OÜ, licensed under the Elastic License 2.0.
// SPDX-License-Identifier: Elastic-2.0

import { Policy, t, typegraph } from "@typegraph/sdk/index.ts";
import { TemporalRuntime } from "@typegraph/sdk/providers/temporal.ts";

export const tg = await typegraph("temporal", (g: any) => {
  const pub = Policy.public();
  const temporal = new TemporalRuntime({
    name: "test",
    hostSecret: "HOST",
    namespaceSecret: "NAMESPACE",
  });
  g.expose({
    startKv: temporal
      .startWorkflow("keyValueStore", t.struct({}))
      .withPolicy(pub),

    query: temporal
      .queryWorkflow("getValue", t.string(), t.string().optional())
      .withPolicy(pub),

    signal: temporal
      .signalWorkflow(
        "setValue",
        t.struct({ key: t.string(), value: t.string() }),
      )
      .withPolicy(pub),

    describe: temporal.describeWorkflow().withPolicy(pub),
  });
});
